services/markup/processorhelper.go

```diff
@@ -8,22 +8,24 @@ import (
 	"context"
 
 	"code.gitea.io/gitea/models/user"
-	"code.gitea.io/gitea/modules/log"
+	module_context "code.gitea.io/gitea/modules/context"
 	"code.gitea.io/gitea/modules/markup"
 )
 
 func ProcessorHelper() *markup.ProcessorHelper {
 	return &markup.ProcessorHelper{
 		IsUsernameMentionable: func(ctx context.Context, username string) bool {
-			// TODO: cast ctx to modules/context.Context and use IsUserVisibleToViewer
-
-			// Only link if the user actually exists
-			userExists, err := user.IsUserExist(ctx, 0, username)
+			mentionedUser, err := user.GetUserByName(ctx, username)
 			if err != nil {
-				log.Error("Failed to validate user in mention %q exists, assuming it does", username)
-				userExists = true
+				return false
+			}
+
+			moduleCtx, ok := ctx.(*module_context.Context)
```

Comment:
It will fail the unit test.

The unit test should construct a &Context{}.

Reply:
Like this? https://github.com/go-gitea/gitea/pull/21554/commits/08bdc8b9d55839b2b715db0ad5467c98d390a897

Reply:
Maybe you need to choose a public user and a private user, and fill the user to `Doer` to let the test check whether `IsUsernameMentionable -> IsUserVisibleToViewer` works as expected.

Reply:
Something's wrong with DB access
```
code.gitea.io/gitea/modules/context.(*Context).Value(0xc001837ed0?, {0x17013e0?, 0x31975b0?})
        /workspace/gitea/modules/context/context.go:574 +0x5e
code.gitea.io/gitea/models/db.GetEngine({0x216d368, 0xc001a8aea0})
        /workspace/gitea/models/db/context.go:78 +0x57
```

Reply:
You need a mocked context, I did some fixes for it